src/irc.py

```diff
@@ -45,9 +46,28 @@ class IRCBRidgeService:
     The app registration file will be used to register the bridge with the Matrix homeserver.
     PEM and the configuration file will be used by the matrix-appservice-irc service.
     """
+    def __init__(self, charm: CharmBase) -> None:
+        super().__init__(charm)
+        self._charm = charm
```

Comment:
The service must not use the charm in order to keep a separation of concerns.
The service can use:
- Pebble (irrelevant here)
- Any service instantiated by the charm

Reply:
I need this for the config, so a `juju config irc enable_ident=true` will change the value in the config file, should I just pass the config instead? I didn't want to go the route of injecting and passing a charm state, it seemed complex for this charm.